pkg/types/clusterconfig/clusterconfig.go

```diff
@@ -524,48 +508,6 @@ func CheckSpotInstancePriceCompatibility(target aws.InstanceMetadata, suggested
 	return nil
 }
 
-func CompatibleSpotInstances(awsClient *aws.Client, targetInstance aws.InstanceMetadata, maxPrice *float64, numInstances int) []aws.InstanceMetadata {
-	compatibleInstances := []aws.InstanceMetadata{}
-	instanceMap := aws.InstanceMetadatas[targetInstance.Region]
-	availableInstances := []aws.InstanceMetadata{}
-
-	for instanceType, instanceMetadata := range instanceMap {
-		if instanceType == targetInstance.Type {
-			continue
-		}
-		availableInstances = append(availableInstances, instanceMetadata)
-	}
-
-	sort.Slice(availableInstances, func(i, j int) bool {
-		return availableInstances[i].Price < availableInstances[j].Price
-	})
-
-	for _, instanceMetadata := range availableInstances {
-		if err := CheckCortexSupport(instanceMetadata); err != nil {
-			continue
-		}
-
-		if err := CheckSpotInstanceCompatibility(targetInstance, instanceMetadata); err != nil {
-			continue
-		}
-
-		spotInstancePrice, awsErr := awsClient.SpotInstancePrice(instanceMetadata.Region, instanceMetadata.Type)
-		if awsErr == nil {
-			if err := CheckSpotInstancePriceCompatibility(targetInstance, instanceMetadata, maxPrice, spotInstancePrice); err != nil {
-				continue
-			}
-		}
-
-		compatibleInstances = append(compatibleInstances, instanceMetadata)
-
-		if len(compatibleInstances) == numInstances {
-			break
-		}
-	}
-
-	return compatibleInstances
-}
-
 func AutoGenerateSpotConfig(awsClient *aws.Client, spotConfig *SpotConfig, region string, instanceType string) error {
 	chosenInstance := aws.InstanceMetadatas[region][instanceType]
```

Comment:
nit: this also could be `primaryInstance` or `primaryInstanceType`

Reply:
Agreed as well.